### PR TITLE
HPCC-26170 Add AuthenticateFeature Transform to envmod templates

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+                    {
+                        "name": "service",
+                        "value": "ws_workunits"
+                    }
+                ]
+            }
+        },
+        {
+            "action": "create",
+            "description": "Add the AuthenticateFeature key to all ESP bindings for ECL Watch",
```
Comment (kenrowland) [inline initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json:76]: Just for correctness the description should say ESP services, not bindings.

File: initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                            "name": "WsSQLName",
+                            "duplicate_ok": true
+                        }
+                    }
+                ]
+            }
+        },
+        {
+            "action": "create",
+            "description": "Add the AuthenticateFeature key to all ESP bindings for ECL Watch",
```
Comment (kenrowland) [inline initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json:118]: Another nit, this is technically the work units service (I think)

File: initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                    {
+                        "name": "service",
+                        "value": "ws_sql"
+                    }
+                ]
+            }
+        },
+        {
+            "action": "create",
+            "description": "Add the AuthenticateFeature key to all ESP bindings for ECL Watch",
```
Comment (kenrowland) [inline initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json:144]: And, this is adding the feature to the build set entry for the ECL Watch service (not the binding)

**Review by kenrowland — COMMENTED** [review]
@Michael-Gardner Looks good. Just a couple of minor comment changes in case someone is reading the template. Also, I want to make sure you've attempted to run this against a production environment and confirmed the access is granted.

**Comment by ghalliday** [comment]
@Michael-Gardner this is waiting for your response...

File: initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json (hunk 2 above)
Comment (Michael-Gardner) [reply initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json]: Changed to work unit service

File: initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json (hunk 3 above)
Comment (Michael-Gardner) [reply initfiles/componentfiles/configschema/templates/AddAuthenticateFeature.json]: Changed to service

**Comment by Michael-Gardner** [comment]
@kenrowland I doubt we're ever going to get that old environment file. But I have tested it against one I created using the Jira information and it does the transforms correctly.

**Pull request merged**